From 1c967ad272611a6601b509eafa73489fd9c9b208 Mon Sep 17 00:00:00 2001
From: John Howard <howardjohn@google.com>
Date: Wed, 9 Mar 2022 14:35:35 -0800
Subject: Move managed gateway our of loop (#37837)

Currently its inside the Ingresses loop; but we are deploying a stand
alone gateway, so it doesn't make sense to deploy it 3 times
---
 tests/integration/pilot/ingress_test.go | 30 ++++++++++++-------------
 1 file changed, 15 insertions(+), 15 deletions(-)

diff --git a/tests/integration/pilot/ingress_test.go b/tests/integration/pilot/ingress_test.go
index 124d6cb5b7..e0ea3860bc 100644
--- a/tests/integration/pilot/ingress_test.go
+++ b/tests/integration/pilot/ingress_test.go
@@ -242,8 +242,9 @@ func TestGateway(t *testing.T) {
 						})
 					})
 				})
-				t.NewSubTest("managed").Run(func(t framework.TestContext) {
-					t.ConfigIstio().YAML(`apiVersion: gateway.networking.k8s.io/v1alpha2
+			}
+			t.NewSubTest("managed").Run(func(t framework.TestContext) {
+				t.ConfigIstio().YAML(`apiVersion: gateway.networking.k8s.io/v1alpha2
 kind: Gateway
 metadata:
   name: gateway
@@ -267,20 +268,19 @@ func TestGateway(t *testing.T) {
     - name: b
       port: 80
 `).ApplyOrFail(t, apps.Namespace.Name())
-					apps.PodB[0].CallOrFail(t, echo.CallOptions{
-						Port:   echo.Port{ServicePort: 80},
-						Scheme: scheme.HTTP,
-						HTTP: echo.HTTP{
-							Headers: headers.New().WithHost("bar.example.com").Build(),
-						},
-						Address: fmt.Sprintf("gateway.%s.svc.cluster.local", apps.Namespace.Name()),
-						Check:   check.OK(),
-						Retry: echo.Retry{
-							Options: []retry.Option{retry.Timeout(time.Minute)},
-						},
-					})
+				apps.PodB[0].CallOrFail(t, echo.CallOptions{
+					Port:   echo.Port{ServicePort: 80},
+					Scheme: scheme.HTTP,
+					HTTP: echo.HTTP{
+						Headers: headers.New().WithHost("bar.example.com").Build(),
+					},
+					Address: fmt.Sprintf("gateway.%s.svc.cluster.local", apps.Namespace.Name()),
+					Check:   check.OK(),
+					Retry: echo.Retry{
+						Options: []retry.Option{retry.Timeout(time.Minute)},
+					},
 				})
-			}
+			})
 		})
 }
 
-- 
2.35.3

